### DISCUSSION TITLE
Disconnect Bluetooth

### DISCUSSION THREAD

**Discussion by bmvd** [pschatzmann/ESP32-A2DP#504] [Q&A] [answered]
I am currently trying to build a WiFi Radio and Bluetooth speaker in the arduino IDE, with the aim of being able to switch between both modes.

Is there a way to completely "shut down bluetooth" instead of just disconnecting from the current device?

It appears that bluetooth does not completely shut down after disconnecting, causing the WiFi part of the equation not to work.

**Comment by pschatzmann** [comment, marked as answer]
I think this has already been answered several times!
When ending a2dp you need to decide if you want it to be able to start a2dp again or not. 
The most flexible way is therefore just to restart...
